app/services/container_deployment_service.rb

```diff
@@ -0,0 +1,65 @@
+class ContainerDeploymentService
+  def all_data
+    {
+      :provision              => possible_provision_providers,
+      :providers              => possible_providers_and_vms,
+      :cloud_init_template_id => cloud_init_template_id,
+    }.compact
+  end
+
+  def possible_provision_providers
+    result = []
+    providers = ExtManagementSystem.all.select do |m|
+      m.instance_of?(ManageIQ::Providers::Amazon::CloudManager) || m.instance_of?(ManageIQ::Providers::Redhat::InfraManager)
+    end
+    providers.each do |provider|
+      result << {:provider => provider, :templates => templates(provider.miq_templates)}
+    end
+    result
```

Comment:
if you replace ```providers.each``` with ```providers.map``` you should be able to drop the first and last lines of this method